packages/client/hmi-client/src/components/widgets/tera-context-menu.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+	<Portal :appendTo="props.appendTo">
+		<div class="tera-context-menu" ref="container">
+			<Menu v-if="isVisible" :model="props.model" />
+		</div>
+	</Portal>
+</template>
+
+<script setup lang="ts">
+import { ref, onUnmounted, PropType } from 'vue';
+import Portal from 'primevue/portal';
+import Menu from 'primevue/menu';
+import { MenuItem } from 'primevue/menuitem';
+import { DomHandler, ZIndexUtils } from 'primevue/utils';
+
+const props = defineProps({
+	model: {
+		type: Array as PropType<MenuItem[]>,
+		default: null
+	},
+	appendTo: {
+		type: String,
+		required: false,
+		default: 'body'
+	},
+	baseZIndex: {
+		type: Number,
+		required: false,
+		default: 0
+	}
+});
```

Comment:
We have quite a few z-index issues already. 
Is this really a good idea to give a control here when you don't really have control? 
e.g. z-index is not absolutely comparative with each other....z-index of 10 in the side-panel is different than z-index of 10 in the header.

Why do we need a wrapper component anyway?

Reply:
we need a wrapper component because the `primevue/contextmenu` does as shown below and it doesnt seem to have an option to use headers instead of nested submenus

<img width="465" alt="Screenshot 2023-06-26 at 12 08 57 PM" src="https://github.com/DARPA-ASKEM/Terarium/assets/14062458/f10449d9-a114-419a-ba16-8ddb7309b0c7">